Framework/Algorithms/test/ExtractMaskTest.h

```diff
@@ -59,6 +60,67 @@ class ExtractMaskTest : public CxxTest::TestSuite {
     AnalysisDataService::Instance().remove(outputWS->getName());
   }
 
+  void test_That_Masked_Detector_List_Populated_When_Passed_A_Mask_Workspace() {
+    // Create a simple test workspace
+    const int nvectors(50), nbins(10);
+    Workspace2D_sptr inputWS =
+        WorkspaceCreationHelper::create2DWorkspace(nvectors, nbins);
+    // Mask every 10th spectra
+    std::set<int64_t> maskedIndices;
+    for (int i = 0; i < 50; i += 10) {
+      maskedIndices.insert(i);
+    }
+    // A few randoms
+    maskedIndices.insert(5);
+    maskedIndices.insert(23);
+    maskedIndices.insert(37);
+    inputWS = WorkspaceCreationHelper::maskSpectra(inputWS, maskedIndices);
+    const std::string inputName("inputWSMask");
+    AnalysisDataService::Instance().add(inputName, inputWS);
+    MaskWorkspace_sptr inputWSMask = runExtractMask(inputName);
+    std::vector<Mantid::detid_t> expectedDetectorList = {1,  6,  11, 21,
+                                                         24, 31, 38, 41};
+
+    std::vector<Mantid::detid_t> detectorList;
+
+    TS_ASSERT_THROWS_NOTHING(
+        detectorList = runExtractMaskReturnList(inputWSMask->getName()));
+    TS_ASSERT_EQUALS(detectorList, expectedDetectorList)
```

Comment:
Spontaneously I would check sizes match and then loop over the elements of the vectors?

Reply:
That's cool, I did not know it.